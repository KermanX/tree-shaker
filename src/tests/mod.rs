use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
