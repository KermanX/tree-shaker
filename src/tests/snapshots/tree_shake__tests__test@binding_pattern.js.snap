---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/binding_pattern.js
---
export function simple(unknown) {
	let { a } = { a: 1 };
	effect(a);
	let { b: c } = { b: 2 };
	effect(c);
	let { d = 3 } = { d: undefined };
	effect(d);
	let { [0 + 1]: e } = { [1]: 4 };
	effect(e);
	let { g: { h: __unused_83AC, i: { j: __unused_C4E6 } } } = unknown;
}
export function with_rest(unknown) {
	let { a: __unused_AA1B,...rest } = {
		a: 1,
		b: unknown,
		c: 2
	};
	effect(rest.a, rest.b, rest.c);
	let { b: __unused_4486,...__unused_2075 } = unknown;
	let { d } = {
		d: unknown,
		e: unknown
	};
	effect(d);
}
